test/001_highsnr_stdio_audio/test_multimode_rx.py

```diff
@@ -0,0 +1,161 @@
+    quit()
+
+N_BURSTS = args.N_BURSTS
+N_FRAMES_PER_BURST = args.N_FRAMES_PER_BURST
+AUDIO_INPUT_DEVICE = args.AUDIO_INPUT_DEVICE
+TIMEOUT = args.TIMEOUT
+
+# AUDIO PARAMETERS
+AUDIO_FRAMES_PER_BUFFER = 2048 
+MODEM_SAMPLE_RATE = codec2.api.FREEDV_FS_8000
+AUDIO_SAMPLE_RATE_RX = 48000
+assert (AUDIO_SAMPLE_RATE_RX % MODEM_SAMPLE_RATE) == 0
+
+# SET COUNTERS
+rx_total_frames_datac0 = 0
+rx_frames_datac0 = 0
+rx_bursts_datac0 = 0
+
+rx_total_frames_datac1 = 0
+rx_frames_datac1 = 0
+rx_bursts_datac1 = 0
+
+rx_total_frames_datac3 = 0
+rx_frames_datac3 = 0
+rx_bursts_datac3 = 0
+
+
+# open codec2 instance        
+datac0_freedv = cast(codec2.api.freedv_open(14), c_void_p)
+datac0_bytes_per_frame = int(codec2.api.freedv_get_bits_per_modem_frame(datac0_freedv)/8)
+datac0_n_max_modem_samples = codec2.api.freedv_get_n_max_modem_samples(datac0_freedv)
+datac0_bytes_out = create_string_buffer(datac0_bytes_per_frame * 2)
+codec2.api.freedv_set_frames_per_burst(datac0_freedv,N_FRAMES_PER_BURST)
+datac0_buffer = bytes()
+
+datac1_freedv = cast(codec2.api.freedv_open(10), c_void_p)
+datac1_bytes_per_frame = int(codec2.api.freedv_get_bits_per_modem_frame(datac1_freedv)/8)
+datac1_n_max_modem_samples = codec2.api.freedv_get_n_max_modem_samples(datac1_freedv)
+datac1_bytes_out = create_string_buffer(datac1_bytes_per_frame * 2)
+codec2.api.freedv_set_frames_per_burst(datac1_freedv,N_FRAMES_PER_BURST)
+datac1_buffer = bytes()
+
+datac3_freedv = cast(codec2.api.freedv_open(12), c_void_p)
+datac3_bytes_per_frame = int(codec2.api.freedv_get_bits_per_modem_frame(datac3_freedv)/8)
+datac3_n_max_modem_samples = codec2.api.freedv_get_n_max_modem_samples(datac3_freedv)
+datac3_bytes_out = create_string_buffer(datac3_bytes_per_frame * 2)
+codec2.api.freedv_set_frames_per_burst(datac3_freedv,N_FRAMES_PER_BURST)
+datac3_buffer = bytes()
+
+# check if we want to use an audio device then do an pyaudio init
+if AUDIO_INPUT_DEVICE != -1: 
+    p = pyaudio.PyAudio()
+    # --------------------------------------------OPEN AUDIO CHANNEL RX
+    stream_rx = p.open(format=pyaudio.paInt16,
+                                     channels=1,
+                                     rate=AUDIO_SAMPLE_RATE_RX,
+                                     frames_per_buffer=AUDIO_FRAMES_PER_BUFFER,
+                                     input=True,
+                                     input_device_index=AUDIO_INPUT_DEVICE
+                                     )
+                                     
+
+timeout = time.time() + TIMEOUT
+receive = True
+
+BUF_SIZE = 160
+
+while receive and time.time() < timeout:
+    if AUDIO_INPUT_DEVICE != -1: 
+        buf_size_converted = int(BUF_SIZE*(AUDIO_SAMPLE_RATE_RX/MODEM_SAMPLE_RATE))
+        data_in = stream_rx.read(buf_size_converted,  exception_on_overflow=False)
+        data_in = audioop.ratecv(data_in,2,1,AUDIO_SAMPLE_RATE_RX, MODEM_SAMPLE_RATE, None)
+        data_in = data_in[0].rstrip(b'\x00')
```

Comment:
@DJ2LS - I have a strange bug where I'm getting `len(data_in) = 2` at this point, it's 1920 as expected after L100.

Reply:
BTW, why do we need `rstrip`?